### PR TITLE
CLOUDSTACK-8686:Verify data disk attachment for clusters with only zone wide primary

### PR DESCRIPTION
**Pull request by pritisarap12**
-Attaching the uploaded/allocated volume to a VM on zwps

### CODE REVIEW

**Comment by DaanHoogland** [comment]
I can only judge this on code cleanliness. An added pep8 pull request builder might help;) @pritisarap12 please solicit review from test experts that have access to the required hardware (zwmp?) or add verification instructions for no-knows like me to test this.
Thanks

File: test/integration/testpaths/testpath_attach_disk_zwps.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        return
+
+    @attr(tags=["basic", "advanced"], required_hardware="true")
+    def test_01_attach_datadisk_to_vm_on_zwps(self):
+        """ Attach Data Disk To VM on ZWPS
+            1.   Get zwps storage pool.
+            2.   Attach data disk to vm which is on zwps.
+            3.   Verify disk is attached.
+        """
+
```
Comment (sanju1010) [inline test/integration/testpaths/testpath_attach_disk_zwps.py]: Can you please point me to the code where it gets the zwps?

Comment (pritisarap12) [inline test/integration/testpaths/testpath_attach_disk_zwps.py]: Added missed code to check Zone wide primary storage and adding tags to storage pools with scope="Zone"

**Comment by DaanHoogland** [comment]
All is green and as there is no other reviews and all tests are green, I am going to say: LGTM. But please someone that understands the full ramifications of the test verify. :+1:

File: test/integration/testpaths/testpath_attach_disk_zwps.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            3.  Launch a VM on ZWPS
+            4.  Attach data disk to vm which is on zwps.
+            5.  Verify disk is attached.
+        """
+
+        # Step 1
+        assert len(list(storagePool for storagePool in self.pools
+                        if storagePool.scope == "ZONE")) >= 1,\
+            "There must be at least one zone wide\
+                storage pools available in the setup"
```
Comment (sanju1010) [inline test/integration/testpaths/testpath_attach_disk_zwps.py]: In stead of failing the test if there are no zone wide primary storage in the setup, can you skip the test?

File: test/integration/testpaths/testpath_attach_disk_zwps.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+        # Adding tags to Storage Pools
+        zone_no = 1
+        for storagePool in self.pools:
+            if storagePool.scope == "ZONE":
+                StoragePool.update(
+                    self.apiclient,
+                    id=storagePool.id,
+                    tags=['zwps' + repr(zone_no)])
```
Comment (sanju1010) [inline test/integration/testpaths/testpath_attach_disk_zwps.py]: Please use the code instead of hard coding the tag

Comment (pritisarap12) [inline test/integration/testpaths/testpath_attach_disk_zwps.py]: Updated the testcase:
     -Skip testcase instead for failing it for no Zone wide primary storage found
     -Replace hard-coded values by code for tag.

**Pull request closed without merging**